File: src/main/java/org/ligi/passandroid/model/PassField.java
package org.ligi.passandroid.model;

import android.support.annotation.Nullable;
import java.io.Serializable;
import org.json.JSONException;
import org.json.JSONObject;

public class PassField implements Serializable {

    public final String key;
    public final String label;
    public final String value;

    public PassField(JSONObject jsonObject, AppleStylePassTranslation translation) {
        label = translation.translate(getField(jsonObject, "label"));
        value = translation.translate(getField(jsonObject, "value"));
        key = translation.translate(getField(jsonObject, "key"));
    }

    @Nullable
    private static String getField(final JSONObject object, final String key) {
        if (object.has(key)) {
            try {
                return object.getString(key);
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
        return null;
    }
}

File: android/src/main/java/org/ligi/passandroid/model/PassField.kt
package org.ligi.passandroid.model

class PassField(val key: String?, val label: String?, val value: String?,val hide:Boolean) {

    fun toHtmlSnippet(): String {
        val result=StringBuilder()
        if (label != null) {
            result.append("<b>").append(label).append("</b> ")
        }
        if (value != null) {
            result.append(value)
        }

        if (label != null || value != null) {
            result.append("<br/>")
        }
        return result.toString()
    }
}